### PR TITLE
Add new fields to Order type to show authorize/charge statuses

### PR DESCRIPTION
**Pull request by korycins**
I want to merge this change because:
-  it adds new field to store order total authorized amount 
- it adds new field to store order total charge amount
- it adds new field to store current order authorize status
- it adds new field to store current order charge status
- it adds a filters for new fields (authorizeStatus, chargeStatus)
- It extends type to provide a new fields via API

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-6726 Orders without any transactions should have a PENDING payment status](https://app.clickup.com/t/2549495/SALEOR-6726)

File: saleor/order/models.py [hunk 1]
```diff
@@ -278,23 +303,22 @@ class Meta:
                 fields=["user_email"],
                 opclasses=["gin_trgm_ops"],
             ),
+            GinIndex(
+                name="order_authorize_status",
+                fields=["authorize_status"],
+            ),
+            GinIndex(name="order_charge_status", fields=["charge_status"]),
```
[the remark this answers is not in the document]
Comment (korycins) [reply saleor/order/models.py:310]: Thanks @tomaszszymanski129

File: saleor/order/migrations/0148_add_fields_for_authorize_and_charge.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        total_charged=models.Sum("charged_value", output_field=models.DecimalField())
+    ).values("total_charged")
+    order_ids = TransactionItem.objects.all().values_list("order_id", flat=True)
+
+    Order.objects.filter(id__in=order_ids).update(
+        total_authorized_amount=models.F("total_authorized_amount")
+        + models.Subquery(transaction_items_total_authorized_amount),
+        total_charged_amount=models.F("total_charged_amount")
+        + models.Subquery(transaction_items_total_charged_amount),
+    )
```
Comment (IKarbowiak) [inline saleor/order/migrations/0148_add_fields_for_authorize_and_charge.py:51]: I'm wondering if it does not fail on big databases. 🤔  @tomaszszymanski129 @fowczarek

Comment (maarcingebala) [reply saleor/order/migrations/0148_add_fields_for_authorize_and_charge.py:51]: Yep in some migrations, we used raw SQL to update all orders. @saleor/sre please review migrations in this PR.

File: saleor/order/utils.py [hunk 3]
```diff
@@ -1010,3 +1017,100 @@ def remove_discount_from_order_line(
...
+        sum(order.payments.values_list("captured_amount", flat=True))  # type: ignore
+        or 0
+    )
+    order.total_charged_amount += (
+        sum(
+            order.payment_transactions.values_list(  # type: ignore
+                "charged_value", flat=True
+            )
+        )
+        or 0
```
Comment (maarcingebala) [inline saleor/order/utils.py:1058]: `sum([]) -> 0` so I think this could be skipped.

**Comment by korycins** [comment]
@patrys  can you also take a look at migrations files?

File: saleor/order/migrations/0148_add_fields_for_authorize_and_charge.py (hunk 2 above)
Comment (patrys) [reply saleor/order/migrations/0148_add_fields_for_authorize_and_charge.py:51]: In this case, I don't think Django will fetch any objects, so we may not need to resort to raw SQL.

**Review by maarcingebala — APPROVED** [review]

**Review by fowczarek — APPROVED** [review]

**Pull request merged**